### ISSUE TITLE
INP001:  does not detect implicit namespace packages containing no module

### ISSUE THREAD

**Issue by LoicRiegel** [astral-sh/ruff#28248]
## Reproduce

Create a test package. Inside it, create a namespace package that does not contain any module. For example

```
src
└── test
    ├── a
    └── __init__.p
```
or
```
src
└── test
    ├── a
    │   └── b
    │       ├── __init__.py
    │       └── sub.py
    └── __init__.p
```

**Observed:**

Ruff does not report any "INP" error.

Note that as soon as we create a least one python module in `src/test/a`, ruff reports the INP001 error:
```
> ruff check --select INP
INP001 File `src\test\a\sub.py` is part of an implicit namespace package. Add an `__init__.py`.
--> src\test\a\sub.py:1:1

Found 1 error
```

## Expected behavior

When INP001 is enabled, Ruff should report the missing `test/a/__init__.py,` even when it does not contain any other python module.

I think this aligns with the rule's "philosophy". The [docs](https://docs.astral.sh/ruff/rules/implicit-namespace-package/) says that although namespaces packages are allowed...
> a package that lacks an __init__.py file is typically meant to be a regular package, and the absence of the __init__.py file is probably an oversight

### Version

ruff 0.16.5

**Comment by ntBre** [comment]
I think the second structure should be flagged in preview, although that's not documented. This was added in https://github.com/astral-sh/ruff/pull/14236. As that PR alludes to, the rule runs on files, so it's not really clear to which file we'd attach the diagnostic for a totally empty directory (or which file we'd be analyzing that would initiate the check). It seems reasonable to me to interpret an empty directory, or a directory hierarchy with no Python files whatsoever, as some kind of data directory and not emit diagnostics.

**Comment by LoicRiegel** [comment]
Oh right, it works in preview! I didn't think of it. The PR was merged on November 2024: that's a long time ago, is there a reason it is still in preview? For the docs, should we add a paragraph in the INP001 rule's doc about that?

For the first structure with just an empty directory, you're right, I totally get your point.

**Comment by ntBre** [comment]
Good points! That was quite a while ago. It looks like we previously tried to stabilize in https://github.com/astral-sh/ruff/pull/20269 for 0.13.0 and found in the ecosystem results that it makes the problem in https://github.com/astral-sh/ruff/issues/6474 a bit worse. I thought we were getting close to a fix for that issue, but then I realized that which test framework is being used affects which behavior you need (https://github.com/astral-sh/ruff/issues/6474#issuecomment-5399470227). It looks like your example may also relate to tests?

Either way, I definitely think it makes sense to update the docs to mention the preview behavior.

**Comment by LoicRiegel** [comment]
> It looks like your example may also relate to tests?

No, my example was in the package source, not in the tests. It was really just me forgetting to add an ``__init__.py`` when changing the package layout! And when I found out later, I was surprised that ruff did not complain before, so I opened this issue
